tests/test-sources/plugins/bufferlines/bufferline.nix

```diff
@@ -6,97 +6,114 @@
   example = {
     plugins.bufferline = {
       enable = true;
-      customFilter = ''
-        function(buf_number, buf_numbers)
-          -- filter out filetypes you don't want to see
-          if vim.bo[buf_number].filetype ~= "<i-dont-want-to-see-this>" then
-              return true
-          end
-          -- filter out by buffer name
-          if vim.fn.bufname(buf_number) ~= "<buffer-name-I-dont-want>" then
-              return true
-          end
-          -- filter out based on arbitrary rules
-          -- e.g. filter out vim wiki buffer from tabline in your work repo
-          if vim.fn.getcwd() == "<work-repo>" and vim.bo[buf_number].filetype ~= "wiki" then
-              return true
-          end
-          -- filter out by it's index number in list (don't show first buffer)
-          if buf_numbers[1] ~= buf_number then
-              return true
-          end
-        end
-      '';
-
-      getElementIcon = ''
-        function(element)
-          -- element consists of {filetype: string, path: string, extension: string, directory: string}
-          -- This can be used to change how bufferline fetches the icon
-          -- for an element e.g. a buffer or a tab.
-          -- e.g.
-          local icon, hl = require('nvim-web-devicons').get_icon_by_filetype(opts.filetype, { default = false })
-          return icon, hl
-        end
-      '';
+      settings = {
+        options = {
+          custom_filter = ''
+            function(buf_number, buf_numbers)
+              -- filter out filetypes you don't want to see
+              if vim.bo[buf_number].filetype ~= "<i-dont-want-to-see-this>" then
+                  return true
+              end
+              -- filter out by buffer name
+              if vim.fn.bufname(buf_number) ~= "<buffer-name-I-dont-want>" then
+                  return true
+              end
+              -- filter out based on arbitrary rules
+              -- e.g. filter out vim wiki buffer from tabline in your work repo
+              if vim.fn.getcwd() == "<work-repo>" and vim.bo[buf_number].filetype ~= "wiki" then
+                  return true
+              end
+              -- filter out by it's index number in list (don't show first buffer)
+              if buf_numbers[1] ~= buf_number then
+                  return true
+              end
+            end
+          '';
+          get_element_icon = ''
+            function(element)
+              -- element consists of {filetype: string, path: string, extension: string, directory: string}
+              -- This can be used to change how bufferline fetches the icon
+              -- for an element e.g. a buffer or a tab.
+              -- e.g.
+              local icon, hl = require('nvim-web-devicons').get_icon_by_filetype(opts.filetype, { default = false })
+              return icon, hl
+            end
+          '';
+          separator_style = [
+            "|"
+            "|"
+          ];
+        };
+      };
```

Comment:
This example seems useful (at first glance), can any of it be integrated into the `settingsExample`?

Reply:
Yeah, makes sense. I'll just copy them over so it's more visible.